src/main/boilerplate/spray/json/ProductFormatsInstances.scala.template

```diff
@@ -16,14 +16,21 @@
 
 package spray.json
 
+import scala.collection.mutable.HashSet
+
 trait ProductFormatsInstances { self: ProductFormats with StandardFormats =>
+  def allowExtraKeys: Boolean = true
 [#  // Case classes with 1 parameters
 
   def jsonFormat1[[#P1 :JF#], T <: Product :ClassManifest](construct: ([#P1#]) => T): RootJsonFormat[T] = {
     val Array([#p1#]) = extractFieldNames(classManifest[T])
     jsonFormat(construct, [#p1#])
   }
   def jsonFormat[[#P1 :JF#], T <: Product](construct: ([#P1#]) => T, [#fieldName1: String#]): RootJsonFormat[T] = new RootJsonFormat[T]{
+    val knownFields = HashSet[String]()
```

Comment:
Can we try to make the feature free when `allowExtraKeys` is `false` (default behavior). Maybe just make this a `lazy val` to make sure it is not initialized when not needed.